vault/ui_custom_messages/entry.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package uicustommessages
+
+import (
+	"errors"
+
+	"github.com/hashicorp/go-uuid"
+)
+
+// Entry is a struct that contains a map of custom message ID to Message struct
+// that can be marshalled and unmarshalled using the JSON encoding to/from a
+// slice of byte to translate to a logical.StorageEntry struct. An Entry
+// consists of data for a particular namespace.
+type Entry struct {
+	// Messages is the map of custom message ID to Message struct.
+	Messages map[string]Message `json:"messages"`
+}
+
+// FindMessages searches through all of the custom messages in the receiver
+// Entry struct and only returns those that match the criteria set in the
+// provided FindFilter struct.
+func (e *Entry) FindMessages(filter FindFilter) []Message {
+	result := make([]Message, 0)
+
+	for _, message := range e.Messages {
+		if message.Matches(filter) {
+			result = append(result, message)
+		}
+	}
+
+	return result
+}
+
+// CreateMessage adds a custom message in the receiver Entry struct using the
+// provided Message struct to populate its properties. If the either the
+// start/end times are invalid or the maximum number of messages already exists,
+// then the message is not added.
+func (e *Entry) CreateMessage(message *Message) error {
```

Comment:
Since `CreateMessage` takes a `Message` as an argument, it seems to me like this is more like an `AddMessageToEntry` or similar. What do you think?

Reply:
My intent was to maintain a link between the Operation Callbacks in logical_system_custom_messages.go all the way down into the methods of uicustommessages.Entry.

Reply:
That makes sense, though I don't feel like "Create" is the right verb for this, since it usually takes a Message. I would expect a 'create' method not take a Message as an argument and probably return a Message as a return value. `AddMessage` maybe (since that would work for both the `Manager` and the `Entry`)?